tasks/test.yaml

```diff
@@ -31,6 +32,15 @@ tasks:
       - task: create:k3d-standard-bundle
       - task: deploy:k3d-standard-bundle
       - task: validate-packages
+      - task: compliance:validate
+        with:
+          oscalfile: ./compliance/oscal-component.yaml
+          assessment_results: ./compliance/oscal-assessment-results.yaml
+      - task: compliance:evaluate
+        with:
+          assessment_results: ./compliance/oscal-assessment-results.yaml
```

Comment:
I think it would be helpful to have these pulled into a separate step in the workflow, rather than in the same step. As-is the output gets a bit lost in the 3000+ lines of the install/test (see [here](https://github.com/defenseunicorns/uds-core/actions/runs/9779618133/job/26999348205?pr=496#step:5:3497)).

Related - it would also be great if the evaluation were pulled into a separate workflow/job, using the assessment output. That would enable us to not clearly show the test pipeline as passed even if the compliance pipeline fails. Does that seem feasible/useful?

Reply:
Agreed - will revise

Reply:
updated - pulled each into individual steps. Current state is all in the same workflow - as the assessment-results are temporary artifacts currently - which means we'd need to play the upload/download game. 

Not opposed to revising further.

Reply:
agreed - I wanted to revisit this. Could this be a good space to implement [continue-on-error](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error) for the evaluate step? 

End goal would be to accept the constraint of this step blocking PR/release - but I believe we have enough discovery to do to warrant some soak-testing.

Reply:
I think that would make me more comfortable, and would also resolve the issue with evaluate failing and resulting in the `cp` not happening (still an issue in the branch current state since its a sequential step). Definitely agree with the end goal, don't want to make compliance be an afterthought/"we'll fix that later" but in the near term I think it's the right move to not block on it. 

If we do flip on `continue-on-error` it would be nice to still identify failures clearly (since they won't show up in the action status anymore). What do you think about PR comments on failure? `gh` cli provides an [easy way](https://cli.github.com/manual/gh_pr_comment) to add comments, but there are also [actions that we could leverage](https://github.com/peter-evans/create-or-update-comment) to do more advanced things (i.e. make sure we only comment once + update comment). Don't want to over-engineer on a short-term change (hopefully) but also want to make sure we don't just completely ignore compliance failures because they are out of sight.

Reply:
I didn't want to inject too much in the way of dependencies required to maintain - that said something like that comment action would provide an easy interface for tagging the lula team in to explore the problem and evaluate updates in a non-blocking fashion. 

at a minimum it would provide a stop-gap to addressing compliance failures until another solution was identified that reduced dependencies required. 

Next steps:
- `continue-on-error` to allow failure which unblocks the next step of copying the artifacts
- Implement some process to notify when failure occurred and surface it.

Reply:
`continue-on-error` added alongside a `notify-lula` action. I believe this could be made more generic but I don't have evidence to support that being required currently. 

Workflow exists such that:
- evaluate will always continue-on-error
- if failed - a comment will be made on the pr tagging the @defenseunicorns/lula-dev team
- artifact will always be stored for review
- if the next iteration passes - the comment will be updated to indicate passing
- should continue to update the comment as required until merge